pkg/commands/plan/plan.go

```diff
@@ -322,6 +326,40 @@ func (c *PlanCommand) terraformPlan(ctx context.Context) (*RunResult, error) {
 	stdout.Reset()
 	stderr.Reset()
 
+	util.Headerf(c.Stdout(), "Saving output to JSON file")
+	_, err = c.terraformClient.Show(ctx, &stdout, multiStderr, &terraform.ShowOptions{
+		File:    pointer.To(c.planFilename),
+		NoColor: pointer.To(true),
+		JSON:    pointer.To(true),
+	})
+	if err != nil {
+		return &RunResult{
+			commentDetails: stderr.String(),
+			hasChanges:     hasChanges,
+		}, fmt.Errorf("failed to terraform show: %w", err)
+	}
+
+	planBasename := strings.TrimSuffix(c.planFilename, path.Ext(c.planFilename))
+	jsonFilepath := path.Join(c.childPath, planBasename+".json")
+
+	var data any
+	if err := json.Unmarshal([]byte(stdout.String()), &data); err != nil {
+		return &RunResult{hasChanges: hasChanges}, fmt.Errorf("failed to unmarshal json plan: %w", err)
+	}
+
+	var buf bytes.Buffer
+	if err := json.NewEncoder(&buf).Encode(data); err != nil {
+		return &RunResult{hasChanges: hasChanges}, fmt.Errorf("failed to encode json: %w", err)
+	}
```

Comment:
Is there a reason for the unmarshal and encoding, seems unnecessary? Can't we just write the string bytes to the file?

```go
if err := os.WriteFile("test", []byte(stdout.String()), 0o600); err != nil {
  // handle err
}
```

Reply:
It's to validate that `stdout.String()` is valid JSON, in case we forget to clear it prior to running the `terraform show` command.

Reply:
Seems wasteful, should we just create a new `var jsonOut strings.Builder` to prevent that issue?